### PR TITLE
Merging template updates

### PR DESCRIPTION
**Pull request by christopher-mohr**
Many thanks for contributing to nf-core/epitopeprediction!

Please fill in the appropriate checklist below (delete whatever is not relevant). These are the most common things requested on pull requests (PRs).

## PR checklist
 - [ ] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] If necessary, also make a PR on the [nf-core/epitopeprediction branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/epitopeprediction)
 - [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [x] Make sure your code lints (`nf-core lint .`).
 - [x] Documentation in `docs` is updated
 - [x] `CHANGELOG.md` is updated
 - [x] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/epitopeprediction/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

**Comment by christopher-mohr** [comment]
Not sure where this `test` check is coming from.   

@apeltzer what do you think about the `--mem_mode` parameter? Should we keep it or could we also use `label` in the corresponding processes?

**Comment by apeltzer** [comment]
Label is the preferred way in most of the cases - should probably switch over to that :-)

**Comment by skrakau** [comment]
@christopher-mohr How would you use `label` for this? The goal was that the user can choose from 3 memory settings, depending on the type and size of the input, and does not need to wait for retying.   
Or would you use different labels depending on a user parameter, or omit any user parameter at all?

File: main.nf [hunk 1]
```diff
@@ -46,16 +45,19 @@ def helpMessage() {
       --peptides_split_minchunksize [int]   Used in combination with '--peptides' or '--proteins': minimum number of peptides that should be written into one chunk. Default: 5000
 
     References                              If not specified in the configuration file or you wish to overwrite any of the references
-      --genome [str]                        Specifies the ensembl reference genome version (GRCh37, GRCh38) Default: GRCh37
+      --genome_version [str]                Specifies the ensembl reference genome version (GRCh37, GRCh38) Default: GRCh37
```
Comment (skrakau) [inline main.nf:48]: Why did you change this, to avoid confusion with the genome references? Anyway, this should be added to the CHANGELOG then, see below.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,15 +1,30 @@
...
+
+- [#50](https://github.com/nf-core/epitopeprediction/pull/50) -  Merge template updates (`v1.10.1`, and `v1.10.2`)
+- [#47](https://github.com/nf-core/epitopeprediction/pull/47)  - Update Fred2 to version 2.0.7
+- [#44](https://github.com/nf-core/epitopeprediction/pull/44)   - Check if requested models for specified tools are supported by Fred2
+- [#35](https://github.com/nf-core/epitopeprediction/pull/35) - Merge template updates (`v1.9`)
+
+### `Fixed`
+
+- [#79](https://github.com/nf-core/epitopeprediction/pull/39) - Fix display of prediction tool version [#36](https://github.com/nf-core/epitopeprediction/issues/36)
+
```
Comment (skrakau) [inline CHANGELOG.md:27]:
```suggestion

### `Deprecated`
- [#50 ](https://github.com/nf-core/epitopeprediction/pull/50) - `--genome` is now deprecated, use `--genome_version` instead

```

File: main.nf [hunk 3]
```diff
@@ -20,21 +20,20 @@ def helpMessage() {
...
-    nextflow run nf-core/epitopeprediction -profile <docker/singularity/conda/institute> --input "*.vcf.gz"
+<<<<<<< HEAD
+    nextflow run nf-core/epitopeprediction --input '*.vcf.gz' -profile docker
 
     Mandatory arguments:
-      -profile [str]                        Configuration profile to use. Can use multiple (comma separated)
-                                            Available: conda, docker, singularity, test, awsbatch, <institute> and more
-
-    Input:
       --input [file]                        Path to input data (must be surrounded with quotes)
```
Comment (skrakau) [inline main.nf:27]: This could be confusing as it seems as if the `--input` parameter is mandatory, but it just has to be one of `--input`, `--peptides` or `--proteins`.

File: main.nf [hunk 4]
```diff
@@ -157,12 +160,22 @@ if (workflow.profile.contains('awsbatch')) {
...
+ch_output_docs_images = file("$baseDir/docs/images/", checkIfExists: true)
+
 // Header log info
 log.info nfcoreHeader()
 def summary = [:]
 summary['Pipeline Name']  = 'nf-core/epitopeprediction'
 if(workflow.revision) summary['Pipeline Release'] = workflow.revision
 summary['Run Name']         = custom_runName ?: workflow.runName
+//Standard Params for nf-core pipelines
+summary['Input']            = params.input
```
Comment (skrakau) [inline main.nf:176]:
```suggestion
```
I would delete this here
1) it is already listed below
2) I would only list this if `!params.show_supported_models`

 and move the occurrence below a bit up to the other input options.

File: main.nf [hunk 5]
```diff
@@ -157,12 +160,22 @@ if (workflow.profile.contains('awsbatch')) {
...
+summary['Fasta Ref']        = params.fasta
```
Comment (skrakau) [inline main.nf:177]:
```suggestion
```
this does not exist in this pipeline

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,15 +1,30 @@
...
-- Update Fred2 to version 2.0.7
-- Add parameter `--show_supported_models` to write out information about supported models (alleles, peptide lengths)
-- Add check if requested models for specified tools are supported by Fred2
+## v1.1.0dev - 2020-08-31
 
-## v1.1.0dev - [date]
+### `Added`
+
+- [#45](https://github.com/nf-core/epitopeprediction/pull/45) - Add test for FASTA input
+- [#44](https://github.com/nf-core/epitopeprediction/pull/44) - Add parameter to write out supported models in files
```
Comment (skrakau) [inline CHANGELOG.md:11]:
```suggestion
- [#44](https://github.com/nf-core/epitopeprediction/pull/44) - Add parameter `--show_supported_models` to write out supported models in files
```
maybe it would be nice to directly see which parameters were added?

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,15 +1,30 @@
...
+- [#42](https://github.com/nf-core/epitopeprediction/pull/42) - Add support for FASTA files with protein sequences as input
```
Comment (skrakau) [inline CHANGELOG.md:12]:
```suggestion
- [#42](https://github.com/nf-core/epitopeprediction/pull/42) - Add parameter `--proteins` allowing FASTA files with protein sequences as input
```
same as above: maybe it would be nice to directly see which parameters were added?

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,15 +1,30 @@
...
+- [#31](https://github.com/nf-core/epitopeprediction/pull/31) - Add support for mouse alleles
+- [#30](https://github.com/nf-core/epitopeprediction/pull/30) - Add parameter to change between different memory modes
```
Comment (skrakau) [inline CHANGELOG.md:14]:
```suggestion
- [#30](https://github.com/nf-core/epitopeprediction/pull/30) - Add parameter `--mem_mode` to change between different memory modes
```
same as above: maybe it would be nice to directly see which parameters were added?

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,15 +1,30 @@
...
+- [#29](https://github.com/nf-core/epitopeprediction/pull/29) - Add parallelisation for peptide input
+
+### `Changed`
+
+- [#50](https://github.com/nf-core/epitopeprediction/pull/50) -  Merge template updates (`v1.10.1`, and `v1.10.2`)
+- [#47](https://github.com/nf-core/epitopeprediction/pull/47)  - Update Fred2 to version 2.0.7
+- [#44](https://github.com/nf-core/epitopeprediction/pull/44)   - Check if requested models for specified tools are supported by Fred2
```
Comment (skrakau) [inline CHANGELOG.md:21]: -> maybe this belongs to `Added`?

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,15 +1,30 @@
...
+- [#35](https://github.com/nf-core/epitopeprediction/pull/35) - Merge template updates (`v1.9`)
+
```
Comment (skrakau) [inline CHANGELOG.md:23]:
```suggestion
- [#30](https://github.com/nf-core/epitopeprediction/pull/30) - Set `maxRetries` from 1 to 3

```
I would mention this, as this is a difference in the behaviour (of this specific pipeline) for the user.

File: nextflow.config [hunk 11]
```diff
@@ -37,6 +38,7 @@ params {
   help = false
   show_supported_models = false
   tracedir = "${params.outdir}/pipeline_info"
+  igenomes_base = 's3://ngi-igenomes/igenomes/'
```
Comment (skrakau) [inline nextflow.config:41]:
```suggestion
```
it is not used in this pipeline, right?

File: nextflow.config [hunk 12]
```diff
@@ -16,16 +16,17 @@ params {
...
   wild_type = false
   peptides_split_maxchunks = 100
   peptides_split_minchunksize = 5000
 
   // Additional annotation files
   proteome = false
 
-  // Boilerplate options  
+  // Boilerplate options
+  input = ''
```
Comment (skrakau) [inline nextflow.config:28]:
```suggestion
  input = false
```
should be `false` if not set, is checked in `main.nf`

File: nextflow.config [hunk 13]
```diff
@@ -37,6 +38,7 @@ params {
   help = false
   show_supported_models = false
```
Comment (skrakau) [inline nextflow.config:39]: This is a pipeline specific option, and could be moved to the top

File: nextflow_schema.json [hunk 14]
```diff
@@ -0,0 +1,368 @@
...
+    "description": "A fully reproducible and state of the art epitope prediction pipeline.",
+    "type": "object",
+    "definitions": {
+        "input_output_options": {
+            "title": "Input/output options",
+            "type": "object",
+            "fa_icon": "fas fa-terminal",
+            "description": "Define where the pipeline should find input data and save output data.",
+            "required": [
+                "input",
```
Comment (skrakau) [inline nextflow_schema.json:14]: Not sure, but does this not cause a problem when the user would want to use "peptides" or "proteins" as input?

**Review by skrakau — CHANGES_REQUESTED** [review]

File: main.nf (hunk 1 above)
Comment (christopher-mohr) [reply main.nf:48]: Yes, `--genome` is used for the igenomes in other pipelines, therefore I thought it would make sense to use a different name.

File: nextflow.config (hunk 11 above)
Comment (christopher-mohr) [reply nextflow.config:41]: I remembered that we also kept this in the `hlatyping` pipeline. Can't remember why exactly but I think it doesn't hurt since there is `igenome_ignore`

File: nextflow.config (hunk 12 above)
Comment (christopher-mohr) [reply nextflow.config:28]: `''` evaluates to `false`

File: nextflow_schema.json (hunk 14 above)
Comment (christopher-mohr) [reply nextflow_schema.json:14]: Good point. I'm wondering if we should switch to a single input parameter `input` and flags to specify if it is peptide or protein input (`--peptides`, `--proteins`). What do you think?

File: nextflow.config (hunk 12 above)
Comment (skrakau) [reply nextflow.config:29]: Ah OK good to know, but could it maybe still be used consistently? Either using `''` for all non-boolean variables (and other input parameters) or sticking to `false`?

File: nextflow_schema.json (hunk 14 above)
Comment (skrakau) [reply nextflow_schema.json:14]: I guess this would also make sense to keep consistently for all nf-core pipelines. At least `eager` does it like this too.

Comment (skrakau) [reply nextflow_schema.json:14]: (...as you suggested)

Comment (christopher-mohr) [reply nextflow_schema.json:14]: I will change that in a separate PR.

**Review by skrakau — APPROVED** [review]
Looks great @christopher-mohr !
